src/internal/kopia/filesystem.go

```diff
@@ -0,0 +1,35 @@
+package kopia
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/filesystem"
+
+	"github.com/alcionai/corso/src/pkg/control/repository"
+	"github.com/alcionai/corso/src/pkg/storage"
+)
+
+func filesystemStorage(
+	ctx context.Context,
+	repoOpts repository.Options,
+	s storage.Storage,
+) (blob.Storage, error) {
+	cfg, err := s.StorageConfig()
+	if err != nil {
+		return nil, clues.Stack(err).WithClues(ctx)
+	}
+
+	fsCfg := cfg.(*storage.FilesystemConfig)
+	opts := filesystem.Options{
+		Path: fsCfg.Path,
+	}
+
+	store, err := filesystem.New(ctx, &opts, false)
+	if err != nil {
+		return nil, clues.Stack(err).WithClues(ctx)
```

Comment:
filesystem.New should be expected to attach clues from the ctx already.

Is a `clues.Wrap` more appropropriate than a Stack here? 
```suggestion
		return nil, clues.Stack(err)
```